src/main/java/io/appform/dropwizard/actors/actor/ActorConfig.java

```diff
@@ -131,30 +156,31 @@ public boolean isValidSharding() {
     @ValidationMethod(message = "Custom connection names should be different from default connection names")
     public boolean isCustomConnectionNamesValid() {
 
-        if (Objects.isNull(producer) && Objects.isNull(consumer)) {
+        if (Objects.isNull(producer) && Objects.isNull(consumer) && Objects.isNull(sidelineProcessor)) {
             return true;
         }
 
-        AtomicBoolean validConnectionNames = new AtomicBoolean(true);
-
-        getConnectionNames().forEach(connectionName -> {
-            if (Constants.DEFAULT_CONNECTIONS.contains(connectionName)) {
-                validConnectionNames.set(false);
-            }
-        });
-
-        return validConnectionNames.get();
+        return getConnectionNames().stream()
+                .noneMatch(Constants.DEFAULT_CONNECTIONS::contains);
     }
 
     private Set<String> getConnectionNames() {
-        Set<String> proposedConnectionNames = new HashSet<>();
-        if (Objects.nonNull(producer) && Objects.nonNull(producer.getConnectionIsolationStrategy()) &&
-                producer.getConnectionIsolationStrategy() instanceof SharedConnectionStrategy) {
-            proposedConnectionNames.add(((SharedConnectionStrategy) producer.getConnectionIsolationStrategy()).getName());
+
+        final Set<String> proposedConnectionNames = new HashSet<>();
```

Comment:
Instead of having this legacy iterative validation present before this MR, this validation can be shifted to respective class where only if connection name is provided in SharedConnectionStrategy then that can be matched against DEFAULT_CONNECTIONS. The approach of verifying the same within respective class using javax validation will be cleaner and more optimized compared to legacy approach.